### PR TITLE
feat(Edit Branch View): Allow adding branch paths

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- You can now add paths in the edit branch view

## Test
- Add paths to a branch and make sure it creates the new path correctly. Test for these types of branches
  - Workgroup ID
  - Score
  - Choice Chosen
  - Random

### CODE REVIEW

File: src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts [hunk 1]
```diff
@@ -215,11 +217,20 @@ export class EditBranchComponent extends AbstractBranchAuthoringComponent {
   protected async submit(): Promise<void> {
     this.submitting = true;
     const params = this.getBranchParams();
+    this.createNewPaths(this.branchPaths, params);
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts]: We can get ```this.branchPaths``` in the function?
```suggestion
    this.createNewPaths(params);
```

File: src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts [hunk 2]
```diff
@@ -215,11 +217,20 @@ export class EditBranchComponent extends AbstractBranchAuthoringComponent {
...
     this.updateTransitions(params);
     this.updateTransitionLogic(params);
     this.saveProject();
   }
 
+  private createNewPaths(branchPaths: any[], params: any): void {
+    branchPaths.forEach((path: any, index: number) => {
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts]: Type CreateBranchParams and get ```this.branchPaths``` from class?
```suggestion
  private createNewPaths(params: CreateBranchParams): void {
    this.branchPaths.forEach((path: any, index: number) => {
```

File: src/assets/wise5/services/createBranchService.ts [hunk 3]
```diff
@@ -49,19 +52,38 @@ export class CreateBranchService {
     return newNodeIds;
   }
 
-  private createPathSteps(params: CreateBranchParams, branchNode: any, newNodeIds: string[]): void {
+  createPathSteps(params: CreateBranchParams, branchNode: any, newNodeIds: string[]): void {
```
Comment (hirokiterashima) [inline src/assets/wise5/services/createBranchService.ts]: private?

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Functionality seems to work as described.

In the create branch view, we show the "Add Path" button. We should hide it?

See inline code improvement suggestions. I also wonder if we should break up CreateBranchService into CreateBranchService and EditBranchService? May not be necessary for this PR, but something to consider as we add more functions to the class.

**Comment by geoffreykwan** [comment]
Ok, I've made the changes.

**Review by hirokiterashima — APPROVED** [review]
LGTM

**Pull request merged**